### PR TITLE
feat: adding error to task interface and display it to the user

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding an error property to Task interface

### Screenshot / video of UI

https://github.com/projectatomic/ai-studio/assets/42176370/0d6cedfd-d413-46fd-90f4-7462570ef6bf

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/240

### How to test this PR?

- [x] Unit tests

### CODE REVIEW

File: packages/backend/src/utils/recipeStatusUtils.ts [hunk 1]
```diff
@@ -46,7 +46,22 @@ export class RecipeStatusUtils {
     this.update();
   }
 
-  setTaskState(taskId: string, state: TaskState) {
+  setTaskError(taskId: string, error: string) {
+    if (!this.tasks.has(taskId)) throw new Error('task not found.');
+    const task = this.tasks.get(taskId);
+    this.setTask({
+      // we place the error before, as we do not want to overwrite any existing error (we try to keep the first one)
```
Comment (feloy) [inline packages/backend/src/utils/recipeStatusUtils.ts:53]: Feel free to add some unit tests to guarantee this

Comment (axel7083) [reply packages/backend/src/utils/recipeStatusUtils.ts:53]: Yeah i'm writing them !
> I should have set it as draft sorry

**Review by lstocchi — COMMENTED** [review]
I really like this patch but i was not able to test it.
I tried to edit the ai-config by removing my arch so that there was no containers. I tried to edit the name of the file so that dekstop could not find it. I tried to run it when the podman machine was stopped. 
In all cases nothing happened, I only see the button like i didn't click anything.
Is it me or it does not actually work on those cases?

**Comment by axel7083** [comment]
> I really like this patch but i was not able to test it. I tried to edit the ai-config by removing my arch so that there was no containers. I tried to edit the name of the file so that dekstop could not find it. I tried to run it when the podman machine was stopped. In all cases nothing happened, I only see the button like i didn't click anything. Is it me or it does not actually work on those cases?

Hey! @lstocchi thanks for the feedback and the reminder :smiley: !! I am aware of the issue of nothing being displayed, I wanted to open an issue yesterday and forgot. This is not part of this PR, so I will try to resolve it in a separate one, and fix it. See https://github.com/projectatomic/ai-studio/issues/246 for details. 



> Not sure if it is part of this PR, wen testing with different error cases (invalid yaml for ai-studio.yaml, invalid sources repository), I don't get the error displayed, but the "Application details" window is closed

@feloy I am sure to really get it, the sidepanel is getting closed ? This is _clearly_ not an expected behaviour, do you have some logs/errors in the devtools ? Because I did not really touch code related to the opening/closing of the sidepanel

**Comment by feloy** [comment]
> > I really like this patch but i was not able to test it. I tried to edit the ai-config by removing my arch so that there was no containers. I tried to edit the name of the file so that dekstop could not find it. I tried to run it when the podman machine was stopped. In all cases nothing happened, I only see the button like i didn't click anything. Is it me or it does not actually work on those cases?
> 
> Hey! @lstocchi thanks for the feedback and the reminder 😃 !! I am aware of the issue of nothing being displayed, I wanted to open an issue yesterday and forgot. This is not part of this PR, so I will try to resolve it in a separate one, and fix it. See #246 for details.
> 
> > Not sure if it is part of this PR, wen testing with different error cases (invalid yaml for ai-studio.yaml, invalid sources repository), I don't get the error displayed, but the "Application details" window is closed
> 
> @feloy I am sure to really get it, the sidepanel is getting closed ? This is _clearly_ not an expected behaviour, do you have some logs/errors in the devtools ? Because I did not really touch code related to the opening/closing of the sidepanel

Sorry, not really closed (on the right), but the "> Run Application" is displayed again, as if I didn't click on it, and the list of tasks disappear. Related to the issue #246 ?

**Comment by axel7083** [comment]
> Sorry, not really closed (on the right), but the "> Run Application" is displayed again, as if I didn't click on it, and the list of tasks disappear. Related to the issue #246 ?

Yes https://github.com/projectatomic/ai-studio/pull/247 should fix it :tada: !

**Comment by axel7083** [comment]
Rebased :rocket: !

**Comment by axel7083** [comment]
@lstocchi should be fixed :wrench: !

**Review by feloy — APPROVED** [review]
Works for me 👍

**Comment by lstocchi** [comment]
There is still a missing error when there are no containers for my arch

**Pull request merged**